Use go code to create/delete webhook

ncskier opened the pull request:
# Changes
Fixes #327
Use go code to create/delete webhooks instead of using a TaskRun.
To create/delete the webhook, we subscribe/unsubscribe using the GitHub
PubSubHubbub API. More info about the API here:
https://developer.github.com/v3/repos/hooks/#pubsubhubbub

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

ncskier commented:
/uncc @akihikokuroda 
/uncc @steveodonovan 
/cc @a-roberts 
/cc @dibbles

dibbles commented:
yes - remove the task and also delete using the git client

dibbles commented on webhooks-extension/pkg/endpoints/github.go:
```diff
@@ -0,0 +1,81 @@
+
+	// Enterprise GitHub API URL is "https://github.company.com/api/v3/hub"
+	if company := getGitHubEnterpriseCompany(u); company != "" {
+		hubbubAPI = fmt.Sprintf("https://github.%s.com/api/v3/hub", company)
```
Implies if you use github enterprise your server api must be github.company.com/api/v3  ...... could a company not use any dns hostname they want?  Is this a guideline given by GitHub for GHE installs?  Do we need to consider alternatives?

ncskier replied:
Thanks Duane, good catch! I've updated the code as well as the tests to catch these cases (https://github.com/tektoncd/experimental/pull/351/files#diff-e4756d2adeff9f4613bce3d5106b8b37R108). Please let me know if there's anything else I missed!

ncskier commented:
Thanks for the review Duane! I've removed all instances of that create-webhook Task, and I've updated the git URL processing to catch the GHE case you mentioned.

Also, here's the issue I created for moving from the PubSubHubbub API to the Hooks API: https://github.com/tektoncd/experimental/issues/362.

Let me know if there are any more changes I can make.
